### PR TITLE
[CALCITE-4673] Correlate variables deduplication is not triggered when custom table function presented.

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java [hunk 1]
```diff
@@ -1255,11 +1255,11 @@ private void substituteSubQuery(Blackboard bb, SubQuery subQuery) {
       //
       // If there is no correlation, the expression is replaced with a
       // boolean indicating whether the sub-query returned 0 or >= 1 row.
-      call = (SqlBasicCall) subQuery.node;
-      query = call.operand(0);
       if (!config.isExpand()) {
         return;
       }
+      call = (SqlBasicCall) subQuery.node;
```
Comment (zinking) [inline core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java:1261]: +1

**Comment by zinking** [comment]
looks good

**Comment by zstan** [comment]
@julianhyde @zabetak Checker failed with no additional info, plz help with it ? locally tests are ok.

**Comment by zabetak** [comment]
> @julianhyde @zabetak Checker failed with no additional info, plz help with it ? locally tests are ok.

@zstan I restarted the jobs to see if it passes but in general you can run the CheckerFramework verifications locally by passing the `-PenableCheckerframework` option. 

`./gradlew -PenableCheckerframework :linq4j:classes :core:classes `

**Comment by julianhyde** [comment]
I'm doing final review. I made some cosmetic changes in https://github.com/julianhyde/calcite/tree/4673-correlated-table and am testing now. Will merge shortly.

**Pull request closed without merging**